### PR TITLE
Bugfix immutable result set

### PR DESCRIPTION
**Pull request by AlexDo1**
Quick bugfix: ImmutableResultSets now always contain the input entry if the entry has no associated_groups.

### CODE REVIEW

**Comment by AlexDo1** [comment]
The plan was to create the branch from `master`, this PR should not contain the ISO19115 commits, I can do the PR again.

File: metacatalog/ext/export_standard/schemas/iso19115/iso19115-2.j2 [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+        {% endfor %}
+    </gmd:parentIdentifier>
+    {% endif %}
+    <gmd:hierarchyLevel>
+        {# always dataset #}
+        <gmd:MD_ScopeCode codeList="http://www.isotc211.org/2005/resources/Codelist/gmxCodelists.xml#MD_ScopeCode" codeListValue="dataset" codeSpace="005">dataset</gmd:MD_ScopeCode>
+    </gmd:hierarchyLevel>
+
+    {# this would be our contact (pointOfContact) #}
+    <gmd:contact xlink:title="DOC/NOAA/NESDIS/NODC/NCDDC&gt; National Coastal Data Development Center (pointOfContact)">
```
Comment (mmaelicke) [inline metacatalog/ext/export_standard/schemas/iso19115/iso19115-2.j2:35]: this needs to be updated

File: metacatalog/ext/export_standard/schemas/iso19115/iso19115-2.j2 [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+                            <gmd:electronicMailAddress>
+                                <gco:CharacterString>alexander.dolich@kit.edu</gco:CharacterString>
+                            </gmd:electronicMailAddress>
+                            <gmd:electronicMailAddress>
+                                <gco:CharacterString>mirko.maelicke@kit.edu</gco:CharacterString>
+                            </gmd:electronicMailAddress>
+                        </gmd:CI_Address>
+                    </gmd:address>
+                    <gmd:hoursOfService>
+                        <gco:CharacterString>Monday - Friday 9:00am - 5:00pm</gco:CharacterString>
```
Comment (mmaelicke) [inline metacatalog/ext/export_standard/schemas/iso19115/iso19115-2.j2:68]: I am not sure, who to put here. 
My suggestion: For the metacatalog package we should get rid of all KIT specific settings and put dummy adresses there. 
Then, there needs to be something like a `template='/home/mytemplates'` attribute, to overwrite the place, where the jinja file can be found. alternatively, we use the same folder and put in a `vforwater.j2`, which we also add to the gitignore.

File: metacatalog/ext/export_standard/schemas/iso19115/iso19115-2.j2 [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+                        <gco:CharacterString>{{version}}</gco:CharacterString>
+                    </edition>
+                    <editionDate>
+                        <gco:DateTime>{{publication.isoformat()}}</gco:DateTime>
+                    </editionDate>
+                    
+                    {# TODO: discuss series #}
+                    {# series? -> EntryGroup.uuid? -> CI_Series #}
+                    {# colletiveTitle? -> EntryGroup.title #}
+                    {# which EntryGroup if there are more than one in Entry.associatedGroups? -> all of them? #}
```
Comment (mmaelicke) [inline metacatalog/ext/export_standard/schemas/iso19115/iso19115-2.j2:132]: yep

File: metacatalog/ext/export_standard/schemas/iso19115/iso19115-2.j2 [hunk 4]
```diff
@@ -0,0 +1,412 @@
...
+                        <gco:CharacterString>{{author.organisation_name}}</gco:CharacterString>
+                    </gmd:organisationName>
+                    <gmd:role>
+                        <gmd:CI_RoleCode codeList="https://data.noaa.gov/resources/iso19139/schema/resources/Codelist/gmxCodelists.xml#CI_RoleCode" codeListValue="author">author</gmd:CI_RoleCode>
+                    </gmd:role>
+                </gmd:CI_ResponsibleParty>
+            </gmd:pointOfContact>
+
+            {# coauthors#}
+            {% if authors %}
```
Comment (mmaelicke) [inline metacatalog/ext/export_standard/schemas/iso19115/iso19115-2.j2:168]: I think a ISO is not valid if the main author is missing, or does this only affect coauthors?

File: metacatalog/ext/export_standard/schemas/iso19115/iso19115-2.j2 [hunk 5]
```diff
@@ -0,0 +1,412 @@
...
+                <gmd:CI_ResponsibleParty>
+                    <gmd:individualName>
+                        {# format: last_name, first_name #}
+                        <gco:CharacterString>{{coauthor.last_name}}, {{coauthor.first_name}}</gco:CharacterString>
+                    </gmd:individualName>
+                    <gmd:organisationName>
+                        <gco:CharacterString>{{coauthor.organisation_name}}</gco:CharacterString>
+                    </gmd:organisationName>
+                    <gmd:role>
+                        <gmd:CI_RoleCode codeList="https://data.noaa.gov/resources/iso19139/schema/resources/Codelist/gmxCodelists.xml#CI_RoleCode" codeListValue="coAuthor">coAuthor</gmd:CI_RoleCode>
```
Comment (mmaelicke) [inline metacatalog/ext/export_standard/schemas/iso19115/iso19115-2.j2:180]: We actually encoded the full `CI_RoleCode` into metacatalog and can use this information here accordingly.
Also, there are roles, that can be hard-coded into the template (The vforwter team as the distributor, I think),

File: metacatalog/ext/export_standard/schemas/iso19115/iso19115-2.j2 [hunk 6]
```diff
@@ -0,0 +1,412 @@
...
+                                <gco:Decimal>{{datasource.spatial_scale.extent}}</gco:Decimal>
+                            </gmd:westBoundLongitude>
+                            <gmd:eastBoundLongitude>
+                                <gco:Decimal>{{datasource.spatial_scale.extent}}</gco:Decimal>
+                            </gmd:eastBoundLongitude>
+                            <gmd:southBoundLatitude>
+                                <gco:Decimal>{{datasource.spatial_scale.extent}}</gco:Decimal>
+                            </gmd:southBoundLatitude>
+                            <gmd:northBoundLatitude>
+                                <gco:Decimal>{{datasource.spatial_scale.extent}}</gco:Decimal>
```
Comment (mmaelicke) [inline metacatalog/ext/export_standard/schemas/iso19115/iso19115-2.j2:308]: Is this correct? you always pass the same data into the west/south lon/lat element...

**Pull request closed without merging**